core/src/com/put/urbantraffic/UrbanTrafficFlowSimulation.java

```diff
@@ -89,21 +75,48 @@ public void create() {
 //
 //        car = new Car(new Node(0, 0), new Node(200, 200), new ArrayList<Node>(Arrays.asList(new Node(0, 0), new Node(0, 100), new Node(0, 200), new Node(100, 200), new Node(200, 200))));
 
-        for(int i = 0; i < amountOfCars; i++){
-            cars.add(city.spawnCar());
-        }
+
+        createSpawnCarArray();
 
         SimulationCore simulation = new SimulationCore();
         simulation.city = city;
-        simulation.epochs = 600;
-        simulation.population = 100;
+        simulation.epochs = SETTINGS.EPOCHS;
+        simulation.population = SETTINGS.POPULATION;
         simulation.numberOfCrossings = city.getCrossings().size();
-        simulation.mutationScale = 100;
-        simulation.initialDeltaRange = 1000;
-        simulation.tournamentSelectionContestants = 2;
+        simulation.mutationScale = SETTINGS.MUTATION_SCALE;
+        simulation.initialDeltaRange = SETTINGS.INITIAL_DELTA_RANGE;
+        simulation.tournamentSelectionContestants = SETTINGS.TOURNAMENT_SELECTION_CONTESTANT;
         simulation.startSimulation();
     }
 
+    private void createSpawnCarArray() {
+        double[] trapeze_area = new double[SETTINGS.ENDING_HOUR];
+        double[] cars_per_hour = new double[SETTINGS.ENDING_HOUR];
+
+        for(int i=SETTINGS.STARTING_HOUR; i<SETTINGS.ENDING_HOUR; i++){
+            trapeze_area[i] = (SETTINGS.TRAFFIC_LEVEL_BY_HOUR[i] + SETTINGS.TRAFFIC_LEVEL_BY_HOUR[i+1])/2;
+        }
+        double trapeze_area_sum = Arrays.stream(trapeze_area).sum();
+        for(int i=SETTINGS.STARTING_HOUR; i<SETTINGS.ENDING_HOUR; i++){
+            cars_per_hour[i] = trapeze_area[i]/trapeze_area_sum*SETTINGS.CARS_QUANTITY;
+        }
```

Comment:
Swoja droga trzeba pomyslec o wyjeciu tego tez z tej klasy, bo to nie nalezy do elementu wizualizacji 🤔

Reply:
pomyslec niekoniecznie teraaz to robic xdxd